pkg/util/pflagaliases_test.go

```diff
@@ -0,0 +1,141 @@
+		aliases := map[string][]string{}
+		util.AddFlagAliasesBool(flags, "some-flag", aliases, "someFlag", "sf")
+		return flags, aliases
+	}
+
+	t.Run("basic", func(t *testing.T) {
+		var someFlagValue bool
+		flags, aliases := setup(&someFlagValue)
+		assert.Nil(t, flags.Parse([]string{"--someFlag", "true"}))
+
+		assert.Equal(t, false, someFlagValue)
+
+		util.ApplyFlagAliases(flags, aliases)
+		assert.Equal(t, true, someFlagValue)
+	})
+
+	t.Run("alt", func(t *testing.T) {
+		var someFlagValue bool
+		flags, aliases := setup(&someFlagValue)
+		assert.Nil(t, flags.Parse([]string{"--sf", "true"}))
+
+		assert.Equal(t, false, someFlagValue)
+
+		util.ApplyFlagAliases(flags, aliases)
+		assert.Equal(t, true, someFlagValue)
+	})
+
+	t.Run("no-opt", func(t *testing.T) {
+		var someFlagValue bool
+		flags, aliases := setup(&someFlagValue)
+		assert.Nil(t, flags.Parse([]string{"--someFlag"}))
+
+		assert.Equal(t, false, someFlagValue)
+
+		util.ApplyFlagAliases(flags, aliases)
+		assert.Equal(t, true, someFlagValue)
+	})
+}
+
+func Test_FlagAliases_slice(t *testing.T) {
+	setup := func(value *[]string) (*pflag.FlagSet, map[string][]string) {
+		flags := pflag.NewFlagSet("flagset", pflag.ContinueOnError)
+		flags.StringSliceVar(value, "some-flag", nil, "usage")
+		aliases := map[string][]string{}
+		util.AddFlagAliasesStringSlice(flags, "some-flag", aliases, "someFlag", "sf")
+		return flags, aliases
+	}
+
+	t.Run("basic", func(t *testing.T) {
+		var someFlagValue []string
+		flags, aliases := setup(&someFlagValue)
+		assert.Nil(t, flags.Parse([]string{"--someFlag", "someValue"}))
+
+		// sanity check, the value should have been parsed into the hidden "someFlag" flag, but not copied into the primary "some-flag" yet
+		assert.Nil(t, someFlagValue)
+
+		util.ApplyFlagAliases(flags, aliases)
+		assert.Equal(t, []string{"someValue"}, someFlagValue)
+	})
+
+	t.Run("alt", func(t *testing.T) {
+		var someFlagValue []string
+		flags, aliases := setup(&someFlagValue)
+		assert.Nil(t, flags.Parse([]string{"--sf", "someValue"}))
+
+		assert.Nil(t, someFlagValue)
+
+		util.ApplyFlagAliases(flags, aliases)
+		assert.Equal(t, []string{"someValue"}, someFlagValue)
+	})
+
+	t.Run("multiple", func(t *testing.T) {
+		var someFlagValue []string
+		flags, aliases := setup(&someFlagValue)
+		assert.Nil(t, flags.Parse([]string{"--someFlag", "someValue", "--someFlag", "secondValue", "--someFlag", "thirdValue"}))
+
+		// sanity check, the value should have been parsed into the hidden "someFlag" flag, but not copied into the primary "some-flag" yet
+		assert.Nil(t, someFlagValue)
+
+		util.ApplyFlagAliases(flags, aliases)
+		assert.Equal(t, []string{"someValue", "secondValue", "thirdValue"}, someFlagValue)
+	})
+
+	t.Run("mixed", func(t *testing.T) {
+		var someFlagValue []string
+		flags, aliases := setup(&someFlagValue)
+		assert.Nil(t, flags.Parse([]string{"--some-flag", "initialValue", "--someFlag", "alias1value", "--sf", "alias2value"}))
```

Comment:
It's cool that this actually works, I really didn't think it would. (this all depends on internal implementation quirks inside survey that we can't change)